pkg/query-service/app/opamp/model/agent.go

```diff
@@ -0,0 +1,286 @@
+package data
+
+import (
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"sync"
+	"time"
+
+	"google.golang.org/protobuf/proto"
+
+	"github.com/open-telemetry/opamp-go/protobufs"
+	"github.com/open-telemetry/opamp-go/server/types"
+)
+
+type AgentStatus int
+
+const (
+	AgentStatusUnknown AgentStatus = iota
+	AgentStatusConnected
+	AgentStatusDisconnected
+)
+
+type Agent struct {
+	ID              string      `json:"agentId" yaml:"agentId" db:"agent_id"`
+	StartedAt       time.Time   `json:"startedAt" yaml:"startedAt" db:"started_at"`
+	TerminatedAt    time.Time   `json:"terminatedAt" yaml:"terminatedAt" db:"terminated_at"`
+	EffectiveConfig string      `json:"effectiveConfig" yaml:"effectiveConfig" db:"effective_config"`
+	CurrentStatus   AgentStatus `json:"currentStatus" yaml:"currentStatus" db:"current_status"`
+	remoteConfig    *protobufs.AgentRemoteConfig
+	Status          *protobufs.AgentToServer
+
+	conn      types.Connection
+	connMutex sync.Mutex
+	mux       sync.RWMutex
+}
+
+func New(ID string, conn types.Connection) *Agent {
+	return &Agent{ID: ID, StartedAt: time.Now(), CurrentStatus: AgentStatusConnected, conn: conn}
+}
+
+// Upsert inserts or updates the agent in the database.
+func (agent *Agent) Upsert() error {
+	agent.mux.Lock()
+	defer agent.mux.Unlock()
+
+	_, err := db.NamedExec(`INSERT OR REPLACE INTO agents (
```

Comment:
I understand that this is done to keep a single row for each agent but here are some of the things that I would like to solve.

There can be a case where the config yaml is the correct yaml, but it is a wrong config for otel collector that it gets failed to be applied on the collector.
* How do I go to a previous version of my config ( an append on the table for configs with each change resulting in a new row should solve this)
* By mistake I made a change I want to go back to the previous version.

Reply:
>There can be a case where the config yaml is the correct yaml, but it is a wrong config for otel collector that it gets failed to be applied on the collector.

When the received config fails for the collector, it does the rollback and sends the server RemoteConfigApplied_Failed response i.e. the effective config will still be the last working one. 

>By mistake I made a change I want to go back to the previous version.

We can maintain a history, I don't see any major issue with that.